common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -653,7 +658,10 @@ def get_problem_html(self, encapsulate=True, demand_hint_text=None, hint_index=0
         # so add 1 to check if others exist.
         should_enable_next_hint = demand_hint_possible and hint_index + 1 < len(demand_hints)
 
-        answer_notification_type, answer_notification_message = self._get_answer_notification()
+        answer_notification_type = None
+        answer_notification_message = None
+        if submit_notification:
```

Comment:
What I meant is how about you pass "submit_notification" to the helper method, so this code can go back to a single one-line assignment.